### DISCUSSION TITLE
why the microstream write performance is not good as DB? microsteam write cost is about 100ms+

### DISCUSSION THREAD

**Discussion by useryq8** [microstream-one/microstream#690] [General]
the original issue is https://github.com/microstream-one/bookstore-demo/issues/10,  the actual write cost is very bad.   
the batch mode can't be used for the actual case is that every user add one book and it needs to be stored into microstream to avoid data lost when app shutdown or machine shutdown accidentally. 

Do we have any best practice when using microstream,   for example StorageEntityCache.Default.setGarbageCollectionEnabled(false);  or set housekeeping to 24housr/15min, then call the issueGarbageCollection explicitly at scheduled time ?

such as, only using in reading scenario, not use in writing scenario(write all thing when app starts, not write in real-time when app is running).![image](https://github.com/microstream-one/microstream/assets/141137124/6612b37f-83d2-43df-aeb9-33f3ff8293a5)
 
arthas(https://github.com/alibaba/arthas) analysis log
![image](https://github.com/microstream-one/microstream/assets/141137124/e658a364-26b9-4d61-bdaa-d8b9a9e1dfac)
@fh-ms  @hg-ms

**Comment by fh-ms** [comment]
This is, I guess, a discussion of best practices for specific use cases.

**Comment by fh-ms** [comment]
`StorageEntityCache.Default.setGarbageCollectionEnabled(false)` disables the GC completely. I wouldn't suggest that. For write-heavy applications, it's better to increase the [budget](https://docs.microstream.one/manual/storage/configuration/housekeeping.html) for the [housekeeping](https://docs.microstream.one/manual/storage/housekeeping.html) process.

If you use MicroStream only for a reading scenario, the housekeeping is idle most of the time, so it doesn't affect the performance anyway.

If you want to achieve a high write throughput, you can either segment your data or use lazy collections. See details [here](https://github.com/eclipse-store/store/discussions/111#discussioncomment-8079997), and results [here](https://github.com/eclipse-store/store/discussions/111#discussioncomment-8081437).

**Reply by useryq8** [reply]
@fh-ms  Thank you!  I have already used lazyArrayList, but the write performance is not good as DB(pg),  When I analyze the write cost by arthas(https://github.com/alibaba/arthas) , it seems that he housekeeping affects the write performance, so I want to disable the GarbageCollection in working time, the call issueGarbageCollection  method in night。    my application is write-heavy app, the write/read ratio is about 1:1
